### PR TITLE
[Hotfix][MOD]: testing multipart upload with failed upload parts

### PR DESCRIPTION
**Pull request by hmaheswa**
polarion: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83589550
hotfix bz: https://bugzilla.redhat.com/show_bug.cgi?id=2262650
failed parts not deleted bz: https://bugzilla.redhat.com/show_bug.cgi?id=2266680

pass logs on 7.1:
http://magna002.ceph.redhat.com/cephci-jenkins/hsm/mod_multipart_hotfix/test_multipart_upload_with_failed_parts_using_s3cmd_and_boto3.console.log_7.1
http://magna002.ceph.redhat.com/cephci-jenkins/hsm/mod_multipart_hotfix/cephci-run-XHFYNI/

fail logs on 5.3z6:
http://magna002.ceph.redhat.com/cephci-jenkins/hsm/mod_multipart_hotfix/test_multipart_upload_with_failed_parts_using_s3cmd_and_boto3.console.log_5.3_z6

sample pass log on 5.3z6 with only 10 iterations:
http://magna002.ceph.redhat.com/cephci-jenkins/hsm/mod_multipart_hotfix/cephci-run-KR7AF8/

### CODE REVIEW

File: rgw/v2/tests/s3_swift/reusable.py [hunk 1]
```diff
@@ -595,6 +596,124 @@ def upload_mutipart_object(
...
+        return
+    part_info = {"PartNumber": part_number, "ETag": part_upload_response["ETag"]}
+    parts_info["Parts"].append(part_info)
+
+
+def test_multipart_upload_failed_parts(
+    rgw_client, s3_object_name, bucket_name, part1_path, part2_path
+):
+    parts_info = {"Parts": []}
+    log.info(f"no of parts: 2")
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/reusable.py:630]: @hmaheswa, do we need format string here?

File: rgw/v2/tests/s3_swift/reusable.py [hunk 2]
```diff
@@ -595,6 +596,124 @@ def upload_mutipart_object(
...
+    if len(parts_info["Parts"]) == part_number:
+        log.info("all parts upload completed")
+        response = rgw_client.complete_multipart_upload(
+            Bucket=bucket_name,
+            Key=s3_object_name,
+            UploadId=mpu["UploadId"],
+            MultipartUpload=parts_info,
+        )
+        log.info(f"complete multipart upload: {response}")
+        log.info("multipart upload complete for key: %s" % s3_object_name)
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/reusable.py:712]: please use format string instead of %s

File: rgw/v2/tests/s3cmd/test_s3cmd.py [hunk 3]
```diff
@@ -338,6 +339,73 @@ def test_exec(config, ssh_con):
         log.info("remove user created")
         reusable.remove_user(user_info[0])
 
+    # CEPH-83589550 - Bug 2262650 - [GSS]Missing Objects in RGW S3 bucket while making GET request - 404 not found error
+    elif config.test_ops.get("test_multipart_upload_with_failed_upload_parts", False):
+        log.info(f"Verify multipart upload with failed upload parts in parallel")
```
Comment (anrao19) [inline rgw/v2/tests/s3cmd/test_s3cmd.py:344]: @hmaheswa, can we remove format string here?

File: rgw/v2/tests/s3_swift/reusable.py (hunk 1 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/reusable.py]: @anrao19 
done, please review again

File: rgw/v2/tests/s3_swift/reusable.py (hunk 2 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/reusable.py]: @anrao19 
done, please review again

File: rgw/v2/tests/s3cmd/test_s3cmd.py (hunk 3 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3cmd/test_s3cmd.py]: @anrao19 
done, please review again

**Review by anrao19 — APPROVED** [review]
LGTM

**Review by ckulal — COMMENTED** [review]
LGTM

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**